### PR TITLE
Make ACL filter configurable for method and path

### PR DESCRIPTION
**Pull request by fmarco76**
If the servlet has different ACLs for different HTTP methods or paths the new implementation allows to define a map to identify the correct ACL.

The MAP will associate a key with the ACL and the key is made as `<method>:<path>`. The method is one of the HTTP defined method (e.g. GET, POST, ...) and the special value of "*" indicate all methods. The path is internal of the servlet path and in case of path parameter these can be identified with "{}". 
An example of entry could be:
```
         key= PUT:/token/{} value= token.read
```

### CODE REVIEW

File: base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java [hunk 1]
```diff
@@ -51,14 +76,37 @@ public abstract class ACLFilter extends HttpFilter {
...
+            final String aclSearch = method + ":" + path;
+            if (aclMap!=null) {
+                String[] aclKeys = aclMap.keySet().stream().filter( key -> {
+                    String keyRegex = key.replaceFirst("\\*", ".*").replace("\\{\\}", "([A-Za-z0-9_\\-]*)");
+                    return aclSearch.matches(keyRegex);
+                } ).toArray(String[]::new);
+                for (String a: aclKeys) {
+                    String[] keyParts = a.split(":");
+                    if (keyParts[0].equals("*")){
+                        acl = aclMap.get(a);
```
Comment (edewata) [inline base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java:100]: Do we need a `break` here?

File: base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java [hunk 2]
```diff
@@ -51,14 +76,37 @@ public abstract class ACLFilter extends HttpFilter {
...
         if(request instanceof HttpServletRequest req &&
                 response instanceof HttpServletResponse resp) {
+            method = req.getMethod();
+            path = req.getPathInfo() != null ? req.getPathInfo() : "";
+            final String aclSearch = method + ":" + path;
+            if (aclMap!=null) {
+                String[] aclKeys = aclMap.keySet().stream().filter( key -> {
+                    String keyRegex = key.replaceFirst("\\*", ".*").replace("\\{\\}", "([A-Za-z0-9_\\-]*)");
+                    return aclSearch.matches(keyRegex);
+                } ).toArray(String[]::new);
```
Comment (edewata) [inline base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java:96]: Not an expert on streams, is it possible to get the map values directly? Something like:
```
String[] acls = aclMap.entrySet().stream()
      .filter(e -> { ... })
      .map(Map.Entry::getValue)
      .toArray(String[]::new);

if (acls.length > 0) {
    acl = acls[0];
}
```

**Review by edewata — APPROVED** [review]
I have some questions, but feel free to update/merge assuming it's sufficiently tested.

File: base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java (hunk 1 above)
Comment (fmarco76) [reply base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java:100]: Actually it is not needed. In the list we could have an element with "*" and a matching path, followed by an entry with matching method and matching path. If we `break` here the more specific entry will not be selected.

File: base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java (hunk 2 above)
Comment (fmarco76) [reply base/server/src/main/java/org/dogtagpki/server/rest/v2/ACLFilter.java]: I have modified the stream sorting the keys in reverse order. In this case, the "*" will always be last and it is possible to collect the first element. In this way the loop among the element is not needed.

**Comment by fmarco76** [comment]
@edewata Thanks!

I have done several tests of this code outside of pki. It is not used for the moment. I will do additional tests with the updated TokenService I am working on,  which will use the ACL map.

**Pull request merged**